index.js

```diff
@@ -48,8 +48,53 @@ class ReadStream extends stream.Readable {
   }
 }
 
+class WriteStream extends stream.Writable {
+  constructor(path) {
+    super({
+      highWaterMark: 4 * 1024 * 1024
+    });
+
+    this._output = null;
+    this._writeRequest = null;
+
+    const pathStr = Memory.allocUtf8String(path);
+    /*
+     * flags = O_WRONLY | O_CREAT (= 0x201)
+     * mode  = 0644               (= 0x1a4)
+     */
+    const fd = open(pathStr, 0x201, 0x1a4);
+    if (fd === -1) {
+      this.emit('error', new Error('Unable to open file'));
```

Comment:
this error emit should probably happen in a process.nextTick as the error might be emitted before someone is able to attach